### PR TITLE
Add a test for cast between datetime and date.

### PR DESCRIPTION
**Pull request by Natim**
https://code.djangoproject.com/ticket/29767#ticket

While using with Annotations and Cast I realized that the casting between datetime and date with SQLite3 was kinda broken.

`Cast("datetime_field", DateField())` is transformed into `CAST("datetime_field" AS TEXT)` rather than for instance `substr(CAST(datetime_field AS TEXT),0,11)`

I was able to use extra to fix that and run the query. However I would like to fix this.

This is especially useful for my usecase where I want to GROUP BY the annotated value.

### CODE REVIEW

**Comment by claudep** [comment]
Natim, could you please create a ticket for that issue?

File: tests/db_functions/comparison/test_cast.py [hunk 1]
```diff
@@ -51,6 +51,22 @@ def test_cast_to_integer(self):
                 numbers = Author.objects.annotate(cast_int=Cast('alias', field_class()))
                 self.assertEqual(numbers.get().cast_int, 1)
 
+    def test_case_from_db_datetime_to_date(self):
+        author = Author.objects.create(name='John Smith', age=45)
+        Fan.objects.create(name='Margaret', age=50, author=author)
+        fans = Fan.objects.annotate(fan_for_day=Cast("fan_since", models.DateField()))
```
Comment (claudep) [inline tests/db_functions/comparison/test_cast.py]: Since `fan_since` is None at this point, the test cannot pass! Same below.

Comment (Natim) [reply tests/db_functions/comparison/test_cast.py]: Ok I need to fix that thanks.

**Comment by Natim** [comment]
Refs https://github.com/django/django/pull/10402

**Pull request closed without merging**